src/DistributedLock.Etcd/EtcdLeaseDistributedLock.cs

```diff
@@ -0,0 +1,198 @@
+using dotnet_etcd.interfaces;
+using Etcdserverpb;
+using Medallion.Threading.Internal;
+using V3Lockpb;
+
+namespace Medallion.Threading.Etcd;
+
+/// <summary>
+/// A distributed lock based on holding an exclusive handle to a lock file. The file will be deleted when the lock is released.
+/// </summary>
+public sealed partial class EtcdLeaseDistributedLock : IInternalDistributedLock<EtcdLeaseDistributedLockHandle>
+{
+    private readonly EtcdClientWrapper _etcdClient;
+
+    private readonly (TimeoutValue duration, TimeoutValue renewalCadence, TimeoutValue minBusyWaitSleepTime,
+        TimeoutValue maxBusyWaitSleepTime) _options;
+
+    public EtcdLeaseDistributedLock(IEtcdClient client, string lockName, Action<EtcdLeaseOptionsBuilder>? options = null)
+    {
+        this.LockName = lockName ?? throw new ArgumentNullException(nameof(lockName));
+        if (lockName.Length == 0) { throw new FormatException($"{nameof(lockName)}: may not have an empty file name"); }
```

Comment:
All `DistributedLock` implementations must support _any_ non-null lock name unless an optional `exactName` parameter is passed (not all providers need to support `exactName`; I don't necessarily recommend supporting it here).

This is done by calling `DistributedLockHelpers.GetSafeName(...)`. See for example how this is handled in `AzureBlobLeaseDistributedLock`. 

We should consider what other restrictions besides non-empty must be accounted for. Is there a max length? Are there any reserved values or invalid characters?